poetry_dockerize_plugin/builder.py

```diff
@@ -131,6 +134,22 @@ def extract_python_version(pyversion: str) -> Optional[str]:
         return None
 
 
+def extract_poetry_version() -> str:
+    poetry_version = "1.7.1"
+    try:
+        poetry_lock_path = Path("poetry.lock")
+        if poetry_lock_path.exists():
+            # serach for "@generated by Poetry <version>" in first line
+            with poetry_lock_path.open("r") as f:
+                first_line = f.readline()
+                match = re.search(r"@generated by Poetry (\d+\.\d+\.\d+)", first_line)
```

Comment:
> would you mind to add a unit test with a real lock file for this detection?

I suggest, we lock the pyproject.toml in the tests folder using poetry itself and pin the poetry version in the workflow?

Reply:
I created a dummy project, but the lock file did not get pushed because of the gitignore rules.

However, it will be created in the CI workflow - should it also be included in the repo? 

The tests should run now properly again, at least they did on my fork: https://github.com/the-working-rene/poetry-dockerize-plugin/actions/runs/11104991328/job/30850201594

Reply:
Alright, included it now. Also removed a not any more used method I added earlier